Author: Daniel Baumann <mail@daniel-baumann.ch>
Description:
 Use fd for stdout instead of device, since the latter may not always exist,
 e.g. in chroots (Closes: #581649).

diff -Naurp plymouth.orig/scripts/plymouth-populate-initrd.in plymouth/scripts/plymouth-populate-initrd.in
--- plymouth.orig/scripts/plymouth-populate-initrd.in
+++ plymouth/scripts/plymouth-populate-initrd.in
@@ -338,7 +338,7 @@ function usage() {
     local output="/dev/stdout"
     local rc=0
     if [ "$1" == "error" ]; then
-        output="/dev/stderr"
+        output=">&2"
         rc=1
     fi
 
@@ -383,14 +383,14 @@ inst ${PLYMOUTH_POLICYDIR}/plymouthd.def
 inst ${PLYMOUTH_CONFDIR}/plymouthd.conf $INITRDDIR
 
 if [ -z "$PLYMOUTH_THEME_NAME" ]; then
-    echo "No default plymouth plugin is set" > /dev/stderr
+    echo "No default plymouth plugin is set" >&2
     exit 1
 fi
 
 PLYMOUTH_MODULE_NAME=$(grep "ModuleName *= *" ${PLYMOUTH_DATADIR}/plymouth/themes/${PLYMOUTH_THEME_NAME}/${PLYMOUTH_THEME_NAME}.plymouth | sed 's/ModuleName *= *//')
 
 if [ ! -f ${PLYMOUTH_PLUGIN_PATH}/${PLYMOUTH_MODULE_NAME}.so ]; then
-    echo "The default plymouth plugin (${PLYMOUTH_MODULE_NAME}) doesn't exist" > /dev/stderr
+    echo "The default plymouth plugin (${PLYMOUTH_MODULE_NAME}) doesn't exist" >&2
     exit 1
 fi
 
diff -Naurp plymouth.orig/scripts/plymouth-set-default-theme.in plymouth/scripts/plymouth-set-default-theme.in
--- plymouth.orig/scripts/plymouth-set-default-theme.in
+++ plymouth/scripts/plymouth-set-default-theme.in
@@ -84,14 +84,14 @@ while [ $# -gt 0 ]; do
 
         -l|--list)
                 if [ -n "$THEME_NAME" ]; then
-                        echo "You can only specify --list or a theme name, not both" > /dev/stderr
-                        echo $(usage) > /dev/stderr
+                        echo "You can only specify --list or a theme name, not both" >&2
+                        echo $(usage) >&2
                         exit 1
                 fi
 
                 if [ $DO_RESET -ne 0 ]; then
-                        echo "You can only specify --reset or --list, not both" > /dev/stderr
-                        echo $(usage) > /dev/stderr
+                        echo "You can only specify --reset or --list, not both" >&2
+                        echo $(usage) >&2
                         exit 1
                 fi
 
@@ -104,14 +104,14 @@ while [ $# -gt 0 ]; do
 
         -r|--reset|default)
                 if [ -n "$THEME_NAME" ]; then
-                        echo "You can only specify --reset or a theme name, not both" > /dev/stderr
-                        echo $(usage) > /dev/stderr
+                        echo "You can only specify --reset or a theme name, not both" >&2
+                        echo $(usage) >&2
                         exit 1
                 fi
 
                 if [ $DO_LIST -ne 0 ]; then
-                        echo "You can only specify --reset or --list, not both" > /dev/stderr
-                        echo $(usage) > /dev/stderr
+                        echo "You can only specify --reset or --list, not both" >&2
+                        echo $(usage) >&2
                         exit 1
                 fi
 
@@ -124,20 +124,20 @@ while [ $# -gt 0 ]; do
 
         *)
                 if [ -n "$THEME_NAME" ]; then
-                        echo "You can only specify one theme at a time" > /dev/stderr
-                        echo $(usage) > /dev/stderr
+                        echo "You can only specify one theme at a time" >&2
+                        echo $(usage) >&2
                         exit 1
                 fi
 
                 if [ $DO_RESET -ne 0 ]; then
-                        echo "You can only specify --reset or a theme name, not both" > /dev/stderr
-                        echo $(usage) > /dev/stderr
+                        echo "You can only specify --reset or a theme name, not both" >&2
+                        echo $(usage) >&2
                         exit 1
                 fi
 
                 if [ $DO_LIST -ne 0 ]; then
-                        echo "You can only specify --list or a theme name, not both" > /dev/stderr
-                        echo $(usage) > /dev/stderr
+                        echo "You can only specify --list or a theme name, not both" >&2
+                        echo $(usage) >&2
                         exit 1
                 fi
 
@@ -163,7 +163,7 @@ if [ $DO_RESET -eq 0 ] && [ $DO_INITRD_R
 fi
 
 if [ `id -u` -ne 0 ]; then
-        echo "This program must be run as root" > /dev/stderr
+        echo "This program must be run as root" >&2
         exit 1
 fi
 
@@ -174,14 +174,14 @@ if [ $DO_RESET -ne 0 ]; then
 fi
 
 if [ ! -e ${PLYMOUTH_DATADIR}/plymouth/themes/${THEME_NAME}/${THEME_NAME}.plymouth ]; then
-        echo "${PLYMOUTH_DATADIR}/plymouth/themes/${THEME_NAME}/${THEME_NAME}.plymouth does not exist" > /dev/stderr
+        echo "${PLYMOUTH_DATADIR}/plymouth/themes/${THEME_NAME}/${THEME_NAME}.plymouth does not exist" >&2
         exit 1
 fi
 
 MODULE_NAME=$(grep "ModuleName *= *" ${PLYMOUTH_DATADIR}/plymouth/themes/${THEME_NAME}/${THEME_NAME}.plymouth | sed 's/ModuleName *= *//')
 
 if [ ! -e ${PLYMOUTH_PLUGIN_PATH}${MODULE_NAME}.so ]; then
-        echo "${PLYMOUTH_PLUGIN_PATH}${MODULE_NAME}.so does not exist" > /dev/stderr
+        echo "${PLYMOUTH_PLUGIN_PATH}${MODULE_NAME}.so does not exist" >&2
         exit 1
 fi
 
